apps/web/src/components/tweetContent.ts

```diff
@@ -0,0 +1,42 @@
+const mentionRegex = /@([a-zA-Z0-9_]{4,15})/g;
+const urlRegex = /https?:\/\/[^\s$.?#].[^\s]*/g;
+
+export type InterfaceChunk = {
+  type: 'text' | 'mention' | 'url';
+  index: number;
+  value: string;
+};
+
+const matchAll = <T extends string>(text: string, type: T, regex: RegExp) => {
+  return [...text.matchAll(regex)].map((match) => ({ type, match }));
+};
+
+export const parseContent = (text: string) => {
```

Comment:
is there a better way?

Reply:
I think a lexer is better, I didn't implement that because maybe it's over-engineering.